filters.py

```diff
@@ -51,25 +80,38 @@ def filterString(self, layer: QgsVectorLayer) -> str:
         Returns:
             str: A layer filter string
         """
-        # ST_DISJOINT does not use spatial indexes, but we can use its opposite "NOT ST_INTERSECTS" which does
         spatial_predicate = f"ST_{Predicate(self.predicate).name}"
+        wkt = self.wkt if not self.bbox else self.boxGeometry.asWkt()
+        srid=self.crs.postgisSrid()
+        layer_srid=layer.crs().postgisSrid()
+
+        if layer.storageType() == SENSORTHINGS_STORAGE_TYPE:
+            reprojected_wkt = reproject_wkt_geometry(wkt, srid, layer_srid)
```

Comment:
A comment why client-side reprojection is necessary (no ST_Transform available for SensorThings I presume?) would be nice.